agent/service.go

```diff
@@ -59,15 +67,40 @@ func (ts *TaskService) Create(ctx context.Context, req *shimapi.CreateTaskReques
 
 	// Do not pass any mounts to runc, everything is already mounted for us
 	req.Rootfs = nil
-	// TODO: handle stdio see: https://github.com/firecracker-microvm/firecracker-containerd/issues/17
-	req.Stdin = ""
-	req.Stdout = ""
-	req.Stderr = ""
+	// handle STDIO
+	ts.io, err = cio.NewFIFOSetInDir(defaultStdioPath, req.ID, req.Terminal)
+	if err != nil {
+		log.G(ctx).WithError(err).Error("error proxying io")
+		return nil, err
+	}
+	req.Stdin = ts.io.Stdin
+	req.Stderr = ts.io.Stderr
+	req.Stdout = ts.io.Stdout
+	var stdin io.ReadWriteCloser
+	if ts.io.Stdin != "" {
+		if stdin, err = fifo.OpenFifo(ctx, ts.io.Stdin, syscall.O_WRONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0700); err != nil {
+			return nil, err
+		}
+	}
+	var stdout io.ReadWriteCloser
+	if ts.io.Stdout != "" {
+		if stdout, err = fifo.OpenFifo(ctx, ts.io.Stdout, syscall.O_RDONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0700); err != nil {
+			return nil, err
+		}
+	}
+	var stderr io.ReadWriteCloser
+	if ts.io.Stderr != "" {
+		if stderr, err = fifo.OpenFifo(ctx, ts.io.Stderr, syscall.O_RDONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0700); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
- Move this to `createFifos() (*cio.FIFOSet, error)` ?
- Also can we add some context to err returns, so we can distinguish which one exactly failed?